src/subscriptions-client.js

```diff
@@ -536,20 +524,22 @@ client.getSubscriptionStatus = function() {
                         if (client.closeRegisterDialog) {
                             client.closeRegisterDialog = false;
                         }
-                        dfd.resolve();
                     }
-                })
-                .catch(ex => {
-                    console.debug(ex);
-                    client.subscriptionStatus.status = "unknown";
-                    client.subscriptionStatus.status_msg = _("Unknown");
-                })
-                .then(() => {
-                    getSubscriptionDetails();
-                    needRender();
                 });
     });
-    return dfd.promise();
+
+    promise
+            .then(() => {
+                getSubscriptionDetails();
+                needRender();
+            })
```

Comment:
Note: This is weird. I kept the original behaviour exactly, as I don't know its original intention; i.e. the promise gets resolved in the middle of activity here, instead of at the end.

Maybe it's time to clean this up as well, but that needs some actual clue what happens here :grin:

Reply:
Hmm I guess this was done like this since commit bacb8f84.